src/openforms/registrations/contrib/zgw_apis/plugin.py

```diff
@@ -38,6 +39,46 @@
 logger = logging.getLogger(__name__)
 
 
+def get_variables_properties_from_submission(
+    submission: Submission, connections: dict[str, Any]
```

Comment:
looking at L51, `connections` does not seem to be a dict, but a list of dicts instead. Perhaps you could define a `TypedDict` subclass to describe the shape?